modyn/supervisor/internal/pipeline_executor/evaluation_executor.py

```diff
@@ -226,33 +228,61 @@ def _launch_evaluations_async(
         tasks: list[Future[StageLog]] = []
         logs = SupervisorLogs()
 
-        def worker_func(eval_req: EvalRequest) -> StageLog:
+        # do batching by (dataset_id, model_id)
+        eval_requests_by_model: dict[tuple[str, int], list[EvalRequest]] = {}
+        for eval_req in eval_requests:
+            key_ = (eval_req.dataset_id, eval_req.id_model)
+            eval_requests_by_model[key_] = eval_requests_by_model.get(key_, []) + [eval_req]
```

Comment:
nit but if you'd want to you could use a defaultdict here and just .append() instead of having the get logic